streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -244,14 +247,25 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             }
         }
 
-        builder.addSubscribedTopicsFromAssignment(
-            activeTasks.values().stream().flatMap(Collection::stream).collect(Collectors.toList()),
-            logPrefix
-        );
-
         changelogReader.transitToRestoreActive();
     }
 
+    private void updateInputPartitionsAndResume(final Task task, final Set<TopicPartition> topicPartitions) {
+        final boolean requiresUpdate = !task.inputPartitions().equals(topicPartitions);
+        if (requiresUpdate) {
+            log.trace("update inputPartitions {} {}", task, topicPartitions);
```

Comment:
Let's log more context here, for example:
```
log.trace("Update task {} inputPartitions: current {}, new {}", task,  task. inputPartitions(), topicPartitions);
```

Reply:
Agree. used proposed log format